astropy/coordinates/angles.py

```diff
@@ -660,7 +674,14 @@ def wrap_angle(self):
 
     @wrap_angle.setter
     def wrap_angle(self, value):
-        self._wrap_angle = Angle(value)
+        if not isinstance(value, Angle):
```

Comment:
This issue accounts for about 5.6% of the time spent in the loop.  So again, small by itself, but adds up...